### ISSUE TITLE
Stable and unstable

### ISSUE THREAD

**Issue by synesthesiam** [rhasspy/rhasspy#7]
The `rhasspy-server` Docker image and Hass.io add-ons should have "stable" and "unstable" versions available. My first thought is to use `rhasspy-server:latest` as the stable version, and have an `unstable` or `beta` tag for users who want to try out new stuff.

Is there any way to do this for the Hass.io add-on? Should we also have `develop` branches for the repos so people can expect `master` to be stable, or just point them to releases?

**Comment by koenvervloesem** [comment]
I think that keeping both a develop and master branch would complicate matters. I would use releases for stable versions and make it clear that master isn't meant for end users.

I have little experience with Hass.io addons, but your approach with the Docker images seems fine.

**Comment by timaschew** [comment]
I agree with release versions for something stable. Using latest in docker is sounds to me always similar as using master in git. You don't see at the first glance to which version it's pointing this makes it sometimes hard for reproducibility.

**Comment by maxbachmann** [comment]
How should the pre release version look in terms of the version number in the VERSION file?
I suppose it could be something like `1.0.0-dev`, `1.0.0-unstable` or `1.0.0-alpha` (I do not really care about which tag we use as long as we use the same one everywhere so it does not confuse people)

regarding Docker Hub I would go with something like
latest -> latest stable version
unstable -> latest unstable version
and then in addition a release for each tag so e.g. for a stable release `1.0.0` and `1.0.0-unstable` for a unstable release

**Comment by koenvervloesem** [comment]
I would use `-dev`, because `-alpha`, `-beta`, `-unstable` give the user specific expectations (that the code is alpha quality, beta quality, unstable, ...)

So after we have released a 1.0.0 version, we bump the version of the source code to 1.0.1-dev. Even if there will never be a 1.0.1 release because we decide to release 1.1.0 after 1.0.0, that will not be a problem because the 1.0.1-dev release number is still smaller than 1.1.0.

I'm following the approach of [semantic versioning](https://semver.org/) here.

I have been thinking about the Docker versions. The `latest` tag would have to refer to the latest released version (e.g. 1.0.0) indeed. But I hesitate to call a release `unstable` (see above). Maybe we can just have a `dev` tag and even create a Docker image with this tag for every commit that succeeds all checks?

**Comment by maxbachmann** [comment]
yes I would definetly go with semantic versioning. And release both all dev and non dev docker containers with their respective tags + a latest tag for the latest released version. Should there be a dev tag for the latest version aswell (latest version no matter whether it is a dev or non dev), so someone developing always gets the latest container?

**Comment by koenvervloesem** [comment]
Giving a released version such as 1.0.0 a dev tag will be confusing. Either way it's not that much of a difference. The commits and corresponding Docker images would be:

- Some development stuff -> released with tags 0.9.0-dev, dev
- VERSION = 1.0.0 -> released with tags 1.0.0, latest
- VERSION = 1.0.1-dev -> released with tags 1.0.1-dev, dev
- Some development stuff -> released with tags 1.0.1-dev, dev
- VERSION = 1.0.1 -> released with tags 1.0.1, latest

So if you always update your container image to dev and we don't put the dev tag on the latest versions, you just update directly from 0.9.0-dev to 1.0.1-dev, but the only difference is the VERSION file, no real development differences. I find this less confusing than seeing a 1.0.0 container image with the dev tag.

@synesthesiam What do you think of this discussion? And about choosing version numbers according to the [semantic versioning](https://semver.org/) specification?

**Comment by maxbachmann** [comment]
@koenvervloesem added support for `-dev` to https://github.com/rhasspy/rhasspy-tag-action. It will output the current release type for the next step, which is either the tag (in our case `dev`), or when there is no tag `release`

**Comment by koenvervloesem** [comment]
Cool! And how would you use this release type that this action outputs?

**Comment by maxbachmann** [comment]
```
name: Bump version
on:
  push:
    branches:
      - master
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - name: Bump version and push tag
      uses: rhasspy/rhasspy-tag-action@master
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      id: bump_tag
    - name: update packages
      if: steps.bump_tag.outputs.new_tag
      run: <package-new-version-and-deploy>
```
(Readme)
you basically set a id for the steps and then you can access them using this id in the next steps ;)
 e.g. in this example as
```
steps.bump_tag.new_tag
steps.bump_tag.current_version
steps.bump_tag.current_release_type
```
